### ISSUE TITLE
Tax rules not being applied

### ISSUE THREAD

**Issue by ravloony** [PrestaShop/PrestaShop#17592]
**Describe the bug**

In my site, certain tax rules are not being applied.

The site is in Austria, and I have a setup where
- DE customers have 2 different tax rules: Standard DE 19% and Foodstuff DE 7%
- FR customers have 2 different tax rules: Standard FR 20% and Foodstuff FR 5.5% 
- AT customers have 2 different tax rules: Standard AT 20% and Foodstuff AT 10%
- All other customers from the EU are under the AT rule
- International customers have no tax rule

Rules are applied based on the delivery address.

If I make a cart, and create an account where the delivery address is in France or Germany, the resulting subtotal has no VAT applied at all, even though the rules exist, and are afaict valid.

In the backoffice, product page, price tab (any product), selecting one of the FR or DE rules in the Select2 tax rules dropdown has the same result: the price including taxes is the same as the price without.

**To Reproduce**

**Screenshots**
If applicable, add screenshots or screenrecords to help explain your problem.

**Additional information**
PrestaShop version: 1.7.6.1
PHP version: 7.1

**Comment by khouloudbelguith** [comment]
Hi @ravloony,

In the BO => International => Localization => Default country is `Austria`?
![image](https://user-images.githubusercontent.com/16067358/74222835-9d4bbf00-4cb5-11ea-9370-853a4acb678e.png)

In the BO => Products detail page => Prices Tab => the Tax selected is for DE customers or for `FR customers`, is it?
=> if your country by default is `Austria`, this is a normal behavior => because
1. `DE customers` is applied for Germany country only => you can change your default country to Germany & check, it will well be applied in the BO
2. `FR customers` is applied for France country only => you can change your default country to France & check, it will be applied in the BO

>If I make a cart, and create an account where the delivery address is in France or Germany, the resulting subtotal has no VAT applied at all, even though the rules exist, and are afaict valid.

The product added to the cart, what is the exact Tax rule applied to this product in the BO?

Thanks!

**Comment by ravloony** [comment]
That is strange to me. I would expect selecting a different tax rule to apply it to the price in the BO. And I just checked on demo.prestashop.com and that is the behaviour I observe. Every tax rule when selected is applied to the price accordingly.

To your question, the tax rule selected by default in the pricing tab in the back end is `AT Standard rate (20%)`. But the user with the cart has an address in France, and the cart has no tax applied.

Given that the demo site works fine, I think there is something wrong with the shop data somewhere, but I can't track it down. Any hints would be helpful.

So far, I have found out that this query https://github.com/PrestaShop/PrestaShop/blob/7633be8294d90148d2bdfe3fd1c2d403e72ec91b/classes/tax/TaxRulesTaxManager.php#L100 returns a result only when the `id_country` is neither France (8 in my case) or Germany (1). This means in turn that this `if` condition is never `true` https://github.com/PrestaShop/PrestaShop/blob/4f5d70f1ae3a9ceb5251a850551e79d8ff62a0d7/src/Adapter/Tax/TaxRuleDataProvider.php#L90 as there are no taxes.

**Comment by ravloony** [comment]
I found (or rather understood) the problem. As I supposed there was a problem with our shop data: there should only be 2 tax rule groups (in our case) - standard and food stuff, and the actual taxes are attached to them.
I attached the FR and DE taxes via the `ps_tax_rule` table and now it works again.
As to why this happened now, without (afaik) us changing the tax rules - some mysteries are not to be understood.
